components/constellation/constellation.rs

```diff
@@ -528,6 +533,25 @@ pub struct InitialConstellationState {
     pub wgpu_image_map: WGPUImageMap,
 }
 
+/// Data needed for EvaluateJavaScript
+struct EvaluateJavaScriptData {
+    /// Embedders can run arbitrary javascript via messages to the constellation.
+    /// We forward these messages to the script thread but need to store the callback function.
+    pub(crate) callbacks: HashMap<embedder_traits::ScriptId, fn(Result<JSValue, JSValueError>)>,
```

Comment:
instead of storing a function, it would probably be nicer to store a `dyn` trait implementation.